druid/src/widget/flex.rs

```diff
@@ -625,6 +626,116 @@ impl<T: Data> Flex<T> {
         let new_child = Child::FlexedSpacer(flex, 0.0);
         self.children.push(new_child);
     }
+
+    /// This function calculates max intrinsic cross axis size of opposite direction. Ie. calculate
+    /// intrinsic width of [`Axis::Vertical`] or intrinsic height of a [`Axis::Horizontal`].
+    /// This function for each child:
+    /// * Computes intrinsic main axis size with infinite cross axis. Then,
+    /// * Computes intrinsic cross axis size with main axis bounded from previous step.
+    ///
+    /// It returns the max of all intrinsic cross axis sizes.
+    fn do_opposite<F1, F2>(
+        &mut self,
+        ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        env: &Env,
+        initial_available_size_on_main_axis: f64,
+        mut f1: F1,
+        mut f2: F2,
+    ) -> f64
+    where
+        F1: FnMut(
+            f64,
+            &mut LayoutCtx,
+            BoxConstraints,
+            &mut WidgetPod<T, Box<dyn Widget<T>>>,
+        ) -> f64,
+        F2: FnMut(
+            f64,
+            &mut LayoutCtx,
+            BoxConstraints,
+            &mut WidgetPod<T, Box<dyn Widget<T>>>,
+        ) -> f64,
+    {
```

Comment:
> instead of do_opposite, can we be a little more descriptive?

Made the change. `measure_cross_intrinsic` is much better than what I had. 

> In the same vein as the earlier comment, f1 and f2 don't tell you much. A more descriptive name would be nice

Correct. Made the change. 

> Also: is it necessary to have different closure types? They look to have the same type constraint?

Right. That's what I had the first time. But compiler says `no two closures, even if identical, have the same type`. They have same signature but what they _do_ differ.